utils/grpcutil.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"net"
+	"regexp"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc"
+)
+
+// getProtoandAdd return protocal and address
+func getProtoandAdd(target string) (string, string) {
```

Comment:
I don't understand the purpose of this function. As far as I know, this all works with unix domain sockets (That's how I test it). Could you explain why this is necessary?